s3_tagger.py

```diff
@@ -80,60 +80,78 @@ def read_csv(csv_location, s3_client):
 
 def tag_object(key, s3_client, s3_bucket, csv_data):
     split_string = key.split("/")
-    if not split_string[-1].endswith("_$folder$"):
-        table_name = split_string[-2]
+    table_name = ""
+    db_name = ""
+    attempt_to_tag = False
+
+    if split_string[-1].endswith("_$folder$"):
+        split_string[-1] = split_string[-1][0:-9]
+
+    if split_string[-2] in csv_data:
+        db_name = split_string[-2]
+        table_name = split_string[-1]
+        attempt_to_tag = True
+
+    elif split_string[-3] in csv_data:
         db_name = split_string[-3]
+        table_name = split_string[-2]
+        attempt_to_tag = True
+
+    elif split_string[-4] in csv_data:
```

Comment:
What is this case for? dbname/tablename/partition/file, means the DB being at -3 is the most it can be, what scen ario requires it to be at -4?

Reply:
A partitioned object with a data file:
"data/db1/tab1/partition1/00000_0"

Partitioned object with no data files hits -3.
"data/db1/tab1/partition1_$folder$"

Reply:
Ah okay, I would have changed the if statement to have put the split_string[-3] in csv_data case first, as its going to be that one 99% of the time (normally). whereas split_string[-2] in csv_data should never happen. (Since implies we just have a file which is the name of the table, which is what the _$folder$ represents. However its not going to slow down the batch that much so no need to reorder.